install-wp-tests.sh

```diff
@@ -1,52 +1,159 @@
+function green {
+  # green text to stdout
+	echo "$@" | sed $'s,.*,\e[32m&\e[m,' | xargs -0 printf
+}
+
+function yellow {
+	# yellow text to stderr
+	echo "$@" | sed $'s,.*,\e[33m&\e[m,' | >&2 xargs -0 printf
+}
 
-CACHEDIR=${CACHEDIR-/tmp}
-CACHEDIR=$(echo $CACHEDIR | sed -e "s/\/$//")
+function red {
+	# red text to stderr
+	echo "$@" | sed $'s,.*,\e[31m&\e[m,' | >&2 xargs -0 printf
+}
+
+# Handle true/false values
+function boolean() {
+  if [[ "$1" =~ ^(true|yes|on|1)$ ]]; then
+    echo "true"
+  elif [[ "$1" =~ ^(false|no|off|0)$ ]]; then
+    echo "false"
+  else
+    red "Error: Invalid boolean value for '$2': '$1'. Must be 'true' or 'false'."
+		exit 1
+  fi
+}
 
-WP_CORE_DIR=${WP_CORE_DIR-$CACHEDIR/wordpress/}
+# Arguments passed to the script directly with defaults.
+DB_NAME="${1:-wordpress_unit_tests}"
+DB_USER="${2:-root}"
+DB_PASS="${3:-}"
+DB_HOST="${4:-localhost}"
+WP_VERSION="${5:-latest}"
+SKIP_DB_CREATE=$(boolean "${6:-false}" "SKIP_DB_CREATE")
+INSTALL_VIP_MU_PLUGINS=$(boolean "${7:-false}" "INSTALL_VIP_MU_PLUGINS")
+INSTALL_OBJECT_CACHE=$(boolean "${8:-true}" "INSTALL_OBJECT_CACHE")
+
+# Environment variables with defaults.
+CACHEDIR=${CACHEDIR:-/tmp}
+CACHEDIR=$(echo "$CACHEDIR" | sed -e "s/\/$//") # Remove trailing slash if present.
+WP_CORE_DIR=${WP_CORE_DIR:-"${CACHEDIR}/wordpress"}
+WP_TESTS_DIR=${WP_TESTS_DIR:-/tmp/wordpress-tests-lib} # Only used with core test suite.
+WP_MULTISITE=${WP_MULTISITE:-0}
+WP_INSTALL_CORE_TEST_SUITE=$(boolean "${WP_INSTALL_CORE_TEST_SUITE:-false}" "WP_INSTALL_CORE_TEST_SUITE")
+INSTALL_WP_TEST_DEBUG=$(boolean "${INSTALL_WP_TEST_DEBUG:-false}" "INSTALL_WP_TEST_DEBUG")
+
+# Allow the script to dump all variables for debugging.
+if [ "$INSTALL_WP_TEST_DEBUG" = "true" ]; then
+	echo "WP_VERSION: ${WP_VERSION}"
+	echo "WP_CORE_DIR: ${WP_CORE_DIR}"
+	echo "CACHEDIR: ${CACHEDIR}"
+	echo "WP_TESTS_DIR: ${WP_TESTS_DIR}"
+	echo "WP_MULTISITE: ${WP_MULTISITE}"
+	echo "DB_NAME: ${DB_NAME}"
+	echo "DB_USER: ${DB_USER}"
+	echo "DB_PASS: ${DB_PASS}"
+	echo "DB_HOST: ${DB_HOST}"
+	echo "SKIP_DB_CREATE: ${SKIP_DB_CREATE}"
+	echo "INSTALL_VIP: ${INSTALL_VIP}"
+	echo "INSTALL_OBJECT_CACHE: ${INSTALL_OBJECT_CACHE}"
+	echo "WP_INSTALL_CORE_TEST_SUITE: ${WP_INSTALL_CORE_TEST_SUITE}"
+fi
 
 # Create the cache directory if it doesn't exist.
-mkdir -p $CACHEDIR
+if [[ ! -d $CACHEDIR ]]; then
+	mkdir -p "$CACHEDIR" || red "Could not create directory $CACHEDIR" && exit 1
+fi
 
 download() {
 	# Check if the file has been downloaded in the last couple of hours.
 	# If it has been, use it instead of downloading it again.
 	if [[ -f $2 ]]; then
-		if test "$(find $2 -mmin -240)"; then
+		if test "$(find "$2" -mmin -240)"; then
+			if [ "$INSTALL_WP_TEST_DEBUG" = "true" ]; then
+				yellow "Using cached $2"
+			fi
+
 			return
 		fi
 	fi
 
-	if [ $(which curl) ]; then
+	if [ "$(which curl)" ]; then
 		curl -f -s "$1" > "$2"
-	elif [ $(which wget) ]; then
+	elif [ "$(which wget)" ]; then
 		wget -nv -O "$2" "$1"
 	fi
```

Comment:
Should we add an error exit here (an `else` in the event that neither `curl` nor `wget` are available?